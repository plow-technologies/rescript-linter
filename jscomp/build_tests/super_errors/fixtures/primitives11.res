type aa = option<string>
type bb = option<int>
type a = option<aa>
type b = option<bb>
let a: a = (Some(Some(5)): b)
