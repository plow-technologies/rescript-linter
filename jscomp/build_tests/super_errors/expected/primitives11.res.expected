
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/primitives11.res[0m:[2m5:13-28[0m

  3 [2mâ”‚[0m type a = option<aa>
  4 [2mâ”‚[0m type b = option<bb>
  [1;31m5[0m [2mâ”‚[0m let a: a = ([1;31mSome(Some(5)): b[0m)
  6 [2mâ”‚[0m 

  This has type: [1;31mb[0m [2m(defined as[0m [1;31moption<bb>[0m[2m)[0m
  But it's expected to have type: [1;33ma[0m [2m(defined as[0m [1;33moption<aa>[0m[2m)[0m
  
  The incompatible parts:
    [1;31mbb[0m [2m(defined as[0m [1;31moption<int>[0m[2m)[0m vs [1;33maa[0m [2m(defined as[0m [1;33moption<string>[0m[2m)[0m
    
    Further expanded:
      [1;31mint[0m vs [1;33mstring[0m
  
  You can convert [1;33mint[0m to [1;33mstring[0m with [1;33mBelt.Int.toString[0m.